src/NServiceBus.Core/Receiving/ReceiveComponent.cs

```diff
@@ -28,6 +28,16 @@ partial class ReceiveComponent
             HostingComponent.Configuration hostingConfiguration,
             PipelineSettings pipelineSettings)
         {
+            hostingConfiguration.Services.AddSingleton(sp =>
+            {
+                var addressResolver = sp.GetRequiredService<ITransportAddressResolver>();
+                return new ReceiveAddresses(
+                    //TODO: should the main address be null on a send-only endpoint?
```

Comment:
What would happen if we don't register the ReceiveAddresses at all for send only endpoints?